SConstruct

```diff
@@ -80,11 +81,15 @@ class FreelanEnvironment(Environment):
             self.install_prefix = self.prefix
             self.bin_install_prefix = self.bin_prefix
 
-        if os.path.basename(self['CXX']) == 'clang++':
-            self.Append(CXXFLAGS=['-Qunused-arguments'])
-            self.Append(CXXFLAGS=['-fcolor-diagnostics'])
-        elif os.path.basename(self['CXX']).startswith('g++'):
-            self.Append(CXXFLAGS=['-Wno-missing-field-initializers'])
+        if self['CXX'] is None:
+          raise SCons.Errors.BuildError(
+            errstr="CXX environment variable not set. Are you missing a C++ compiler?");
```

Comment:
Whatever we decide regarding that error, please make sure you use a 4-spaces based indentation and avoid the `;` at the end of the line: those are no necessary in Python.